Python/tracemalloc.c

```diff
@@ -603,18 +603,39 @@ tracemalloc_realloc(void *ctx, void *ptr, size_t new_size)
 static void
 tracemalloc_free(void *ctx, void *ptr)
 {
+    if (ptr == NULL)
+        return;
+
     PyMemAllocatorEx *alloc = (PyMemAllocatorEx *)ctx;
 
+    alloc->free(alloc->ctx, ptr);
+
+    TABLES_LOCK();
+    REMOVE_TRACE(ptr);
+    TABLES_UNLOCK();
+}
+
+
+static void
+tracemalloc_raw_free(void *ctx, void *ptr)
+{
     if (ptr == NULL)
         return;
 
+    PyMemAllocatorEx *alloc = (PyMemAllocatorEx *)ctx;
+
      /* GIL cannot be locked in PyMem_RawFree() because it would introduce
         a deadlock in _PyThreadState_DeleteCurrent(). */
 
     alloc->free(alloc->ctx, ptr);
 
     TABLES_LOCK();
-    REMOVE_TRACE(ptr);
+    if (tracemalloc_config.tracing) {
+        REMOVE_TRACE(ptr);
+    }
+    else {
+        // gh-128679: tracemalloc.stop() was called by another thread
+    }
```

Comment:
I think we can remove the `else`:
```suggestion
    // gh-128679: If we're not tracing, then tracemalloc.stop() was called by another thread
```